### ISSUE TITLE
[core] decompilation not working for any class in JAR

### ISSUE THREAD

**Issue by hohwille** [skylot/jadx#1077]
**Checks before report**
- [ ] check [latest unstable build](https://bintray.com/skylot/jadx/unstable/_latestVersion#files) - not possible because only windows `exe` is available and I have MacOS (maybe issue already fixed)
- [x] check [Troubleshooting Q&A](https://github.com/skylot/jadx/wiki/Troubleshooting-Q&A) section on wiki
- [x] search existing issues by exception message - similar to #1066 but not in any way related to `BOOT-INF\classes`.

**Describe error**

I can open and browse the structure or JAR files in `jadx-gui` but opening any class file fails with the following error:
```
INFO  - output directory: cli-7.1.0
INFO  - loading ...
ERROR - Failed to convert file: /Users/hohwille/Downloads/cli-7.1.0.jar
java.lang.RuntimeException: Java to dex conversion error, code: 1
 errors:
PARSE ERROR:
com.android.dx.cf.iface.ParseException: class name (javax/xml/bind/ModuleUtil) does not match path (META-INF/versions/9/javax/xml/bind/ModuleUtil.class)
	at com.android.dx.cf.direct.DirectClassFile.parse0(DirectClassFile.java:557)
	at com.android.dx.cf.direct.DirectClassFile.parse(DirectClassFile.java:434)
	at com.android.dx.cf.direct.DirectClassFile.parseToInterfacesIfNecessary(DirectClassFile.java:416)
	at com.android.dx.cf.direct.DirectClassFile.getMagic(DirectClassFile.java:256)
	at com.android.dx.command.dexer.Main.parseClass(Main.java:773)
	at com.android.dx.command.dexer.Main.access$1600(Main.java:85)
	at com.android.dx.command.dexer.Main$ClassParserTask.call(Main.java:1766)
	at com.android.dx.command.dexer.Main.processClass(Main.java:756)
	at com.android.dx.command.dexer.Main.processFileBytes(Main.java:724)
	at com.android.dx.command.dexer.Main.access$1200(Main.java:85)
	at com.android.dx.command.dexer.Main$FileBytesConsumer.processFileBytes(Main.java:1719)
	at com.android.dx.cf.direct.ClassPathOpener.processArchive(ClassPathOpener.java:285)
	at com.android.dx.cf.direct.ClassPathOpener.processOne(ClassPathOpener.java:165)
	at com.android.dx.cf.direct.ClassPathOpener.process(ClassPathOpener.java:143)
	at com.android.dx.command.dexer.Main.processOne(Main.java:678)
	at com.android.dx.command.dexer.Main.processAllFiles(Main.java:575)
	at com.android.dx.command.dexer.Main.runMultiDex(Main.java:365)
	at com.android.dx.command.dexer.Main.runDx(Main.java:286)
	at jadx.plugins.input.javaconvert.DxConverter.run(DxConverter.java:40)
	at jadx.plugins.input.javaconvert.JavaConvertLoader.convertJar(JavaConvertLoader.java:87)
	at jadx.plugins.input.javaconvert.JavaConvertLoader.lambda$processJars$1(JavaConvertLoader.java:41)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1654)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at jadx.plugins.input.javaconvert.JavaConvertLoader.processJars(JavaConvertLoader.java:39)
	at jadx.plugins.input.javaconvert.JavaConvertLoader.process(JavaConvertLoader.java:29)
	at jadx.plugins.input.javaconvert.JavaConvertPlugin.loadFiles(JavaConvertPlugin.java:21)
	at jadx.api.JadxDecompiler.loadInputFiles(JadxDecompiler.java:112)
	at jadx.api.JadxDecompiler.load(JadxDecompiler.java:98)
	at jadx.gui.JadxWrapper.openFile(JadxWrapper.java:46)
	at jadx.gui.ui.MainWindow.lambda$open$0(MainWindow.java:376)
	at jadx.gui.jobs.BackgroundExecutor$TaskWorker.doInBackground(BackgroundExecutor.java:100)
	at jadx.gui.jobs.BackgroundExecutor$TaskWorker.doInBackground(BackgroundExecutor.java:77)
	at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
	at jadx.gui.jobs.BackgroundExecutor.lambda$execute$0(BackgroundExecutor.java:41)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
...while parsing META-INF/versions/9/javax/xml/bind/ModuleUtil.class

1 error; aborting

	at jadx.plugins.input.javaconvert.DxConverter.run(DxConverter.java:46)
	at jadx.plugins.input.javaconvert.JavaConvertLoader.convertJar(JavaConvertLoader.java:87)
	at jadx.plugins.input.javaconvert.JavaConvertLoader.lambda$processJars$1(JavaConvertLoader.java:41)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1654)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at jadx.plugins.input.javaconvert.JavaConvertLoader.processJars(JavaConvertLoader.java:39)
	at jadx.plugins.input.javaconvert.JavaConvertLoader.process(JavaConvertLoader.java:29)
	at jadx.plugins.input.javaconvert.JavaConvertPlugin.loadFiles(JavaConvertPlugin.java:21)
	at jadx.api.JadxDecompiler.loadInputFiles(JadxDecompiler.java:112)
	at jadx.api.JadxDecompiler.load(JadxDecompiler.java:98)
	at jadx.gui.JadxWrapper.openFile(JadxWrapper.java:46)
	at jadx.gui.ui.MainWindow.lambda$open$0(MainWindow.java:376)
	at jadx.gui.jobs.BackgroundExecutor$TaskWorker.doInBackground(BackgroundExecutor.java:100)
	at jadx.gui.jobs.BackgroundExecutor$TaskWorker.doInBackground(BackgroundExecutor.java:77)
	at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
	at jadx.gui.jobs.BackgroundExecutor.lambda$execute$0(BackgroundExecutor.java:41)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```
The JAR I tested with this stacktrace was https://repo1.maven.org/maven2/com/devonfw/cobigen/cli/7.1.0/cli-7.1.0.jar
I am using MacOS (Catalina) and installed jadx as described with `brew install jadx`.

Please note that this seems to be only an error of `jadx-gui`.
I also did use jadx CLI:
```
jadx cli-7.1.0.jar
jadx cli-7.1.0/resources/com/devonfw/cobigen/api/CobiGen.class
```
This properly decomiled the `class` file to `CobiGen/sources/com/devonfw/cobigen/api/CobiGen.java` and the result looks sane. However, the GUI brings a lot of value to jadx so I hope this is going to be fixed.

**Comment by jpstotz** [comment]
The reason is the same as #1033. 
Jadx can only decompile dex code. Regular JAR files are therefore first converted do dex code using Android dx. dx has the requirement that all classes in a JAR are locate in it's class path relative to the root directory, otherwise converting and thus decompilation fails.

In your example the class `javax.xml.bind.ModuleUtil` was found within the JAR. Based on it's name it have to be located in `javax/xml/bind/ModuleUtil.class` however it was found in path `META-INF/versions/9/javax/xml/bind/ModuleUtil.class`.

**Comment by skylot** [comment]
Closed as duplicate of #1066
